src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -104,30 +81,54 @@ class JesBackend extends Backend with LazyLogging {
     }
   }
 
+  sealed trait JesParameter {
+    def name: String
+    def gcs: String
+    def local: Path
+    def parameterType: String
+
+    final val isInput = this.isInstanceOf[JesInput]
```

Comment:
Kinda icky that knowledge of subtypes is embedded here.  How about:

```scala
lazy val isInput = !isOutput
lazy val isOutput = !isInput
```

And then override in the subtypes

```scala
final case class JesInput(override val isInput = true, ...)
final case class JesOutput(override val isOutput = true,...)
```